casper/src/test/scala/coop/rchain/casper/HashSetCasperTest.scala

```diff
@@ -523,6 +536,79 @@ class HashSetCasperTest extends FlatSpec with Matchers {
     node.tearDown()
   }
 
+  it should "allow paying for deploys" in {
+    val node      = HashSetCasperTestNode.standalone(genesis, validatorKeys.head)
+    val (sk, pk)  = Ed25519.newKeyPair
+    val user      = ByteString.copyFrom(pk)
+    val timestamp = System.currentTimeMillis()
+    val phloPrice = 1L
+    val amount    = 847L
+    val sigDeployData = ProtoUtil
+      .sourceDeploy(
+        s"""new retCh in { @"blake2b256Hash"!([0, $amount, *retCh].toByteArray(), "__SCALA__") }""",
+        timestamp,
+        accounting.MAX_VALUE
+      )
+      .withUser(user)
+    val sigData = node.runtimeManager
+      .captureResults(
+        genesis.getBody.getPostState.tuplespace,
+        ProtoUtil.deployDataToDeploy(sigDeployData)
+      )
+      .head
+      .exprs
+      .head
+      .getGByteArray
+    val sig   = Base16.encode(Ed25519.sign(sigData.toByteArray, sk))
+    val pkStr = Base16.encode(pk)
+    val paymentCode =
+      s"""new paymentForward, walletCh in {
+         |  for(faucet <- @"faucet"; pos <- @"proofOfStake"){
+         |    faucet!($amount, "ed25519", "$pkStr", *walletCh) |
+         |    for(@[wallet] <- walletCh) {
+         |      for(@purse <- paymentForward){ pos!("pay", purse, Nil) } |
+         |      @wallet!("transfer", $amount, 0, "$sig", *paymentForward, Nil)
+         |    }
+         |  }
+         |}""".stripMargin
+    val paymentDeployData = ProtoUtil
+      .sourceDeploy(paymentCode, timestamp, accounting.MAX_VALUE)
+      .withPhloPrice(phloPrice)
+      .withUser(user)
+
+    val paymentQuery = ProtoUtil.sourceDeploy(
+      """for(pos <- @"proofOfStake") {
+        |  pos!("lastPayment", "__SCALA__")
+        |}""".stripMargin,
+      0L,
+      accounting.MAX_VALUE
+    )
+
+    val Created(block) = node.casperEff.deploy(
+      paymentDeployData
+    ) *> node.casperEff.createBlock
+    val blockStatus = node.casperEff.addBlock(block)
+    val queryResult = node.runtimeManager
+      .captureResults(
+        block.getBody.getPostState.tuplespace,
+        ProtoUtil.deployDataToDeploy(paymentQuery)
+      )
```

Comment:
This is a pattern I've seen multiple times - _deploy and capture_. Can you factor it out? I've already tried doing that [here](https://github.com/rchain/rchain/pull/1680/files#diff-0d32ac2de7f4f712df4777ff7a04d512R171), and the example usage is [here](https://github.com/rchain/rchain/pull/1680/files#diff-0d32ac2de7f4f712df4777ff7a04d512R63).